Azure/azure-service-operator
# Task#965 Add Vnet to PostgreSQLServer

Closes #965 

**What this PR does / why we need it**:
Add vnet to postgreSqlSever. 

**Special notes for your reviewer**:
Most of the codes are referred from MySqlServer and SqlServer with names changed. 
To test the code change: 
1. Normal testing. 
Create a PostgreSqlServer and Vnet first using the sample yaml files. Remember to change the postgreSqlServer to General Purpose instead of basic. 
Create the postgreSqlServerVnetRule using the sample yaml file. The Vnet rules can be create succesfully under the PostgreSqlServer. Status could be either check from Kubectl or portal. 
2. Abnormal testing. 
Create a postgreSqlserver but with the basic tier. The creation of postgreSqlServerVnetRule will get failure message 'FeatureNotSupportedForEdition'. The controller could handle the error properly. 

**How does this PR make you feel**:
![gif](https://giphy.com/)

**If applicable**:
